### PR TITLE
Enable shared build for PhysicsFS

### PR DESCRIPTION
**Pull request by yipdw**
This pull request enables a shared library build for PhysicsFS.

The following are all true in my MXE instance:

1. test-physfs.exe links and successfully runs under Wine 1.6.
2. In a shared build, only libphysfs.dll is present in `bin` and an import library is present in `lib`
3. In a static build, neither the DLL nor the import library are present

The definition of `$(PKG)_BUILD_SHARED` is largely copied from the existing build definition, with a few tweaks to turn off the static build.  I'd appreciate any suggestions to reduce the copy-paste.

### CODE REVIEW

File: src/physfs.mk [hunk 1]
```diff
@@ -32,6 +32,18 @@ define $(PKG)_BUILD
         -lphysfs -lz
 endef
 
-$(PKG)_BUILD_x86_64-w64-mingw32 =
```
Comment (TimothyGu) [inline src/physfs.mk]: Why was this removed?

File: src/physfs.mk [hunk 2]
```diff
@@ -32,6 +32,18 @@ define $(PKG)_BUILD
...
+define $(PKG)_BUILD_SHARED
+    cd '$(1)' && cmake . \
+        -DCMAKE_TOOLCHAIN_FILE='$(CMAKE_TOOLCHAIN_FILE)' \
+        -DPHYSFS_BUILD_SHARED=TRUE \
+        -DPHYSFS_BUILD_STATIC=FALSE \
+        -DPHYSFS_INTERNAL_ZLIB=FALSE \
+        -DPHYSFS_BUILD_TEST=FALSE \
+        -DPHYSFS_BUILD_WX_TEST=FALSE
```
Comment (TimothyGu) [inline src/physfs.mk]: It might be better if you did something like this:

```make
       $(if $(BUILD_SHARED), \
         -DPHYSFS_BUILD_SHARED=TRUE \
         -DPHYSFS_BUILD_STATIC=FALSE, \
         -DPHYSFS_BUILD_SHARED=FALSE) \
```

Comment (TimothyGu) [inline src/physfs.mk]: (And merged `$(PKG)_BUILD` and `$(PKG)_BUILD_SHARED` together.)

File: src/physfs.mk (hunk 1 above)
Comment (yipdw) [inline src/physfs.mk]: Oh, oops -- stray deletion.  I'll put it back.

File: src/physfs.mk (hunk 2 above)
Comment (yipdw) [inline src/physfs.mk]: Cool, thanks for the tips.  I've amended this PR's commit to incorporate your suggestions.

**Comment by TimothyGu** [comment]
LGTM, thanks!

**Comment by yipdw** [comment]
Thanks for the fast review, merge, and great documentation :+1:
